web/modules/custom/weather_block/src/Controller/WeatherFormController.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Drupal\weather_block\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+
+/**
+ * Controller for displaying the weather form.
+ */
+class WeatherFormController extends ControllerBase {
+
+  /**
+   * Displays the weather form.
+   */
+  public function showForm() {
+    return \Drupal::formBuilder()->getForm('Drupal\weather_block\Form\InputWeather\InputWeather');
```

Comment:
Dependency injection

Reply:
Still no DI
But, also this controller is useless and can be changed to _form in your routing.yml